pkg/http/server_test.go

```diff
@@ -97,3 +97,66 @@ func TestNewServer(t *testing.T) {
 		}
 	}
 }
+
+var tlsCipherSuitesTests = []struct {
+	ciphers    []uint16
+	shouldFail bool
+}{
+	{ciphers: nil, shouldFail: false},
+	{ciphers: defaultCipherSuites, shouldFail: false},
+	{
+		ciphers: []uint16{ // unsupported cipher suites
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,    // Go stack contains (some) countermeasures against timing attacks (Lucky13)
+			tls.TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, // No countermeasures against timing attacks
+			tls.TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,    // Go stack contains (some) countermeasures against timing attacks (Lucky13)
+			tls.TLS_ECDHE_ECDSA_WITH_RC4_128_SHA,        // Broken cipher
+			tls.TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA,     // Sweet32
+			tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,      // Go stack contains (some) countermeasures against timing attacks (Lucky13)
+			tls.TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256,   // No countermeasures against timing attacks
+			tls.TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,      // Go stack contains (some) countermeasures against timing attacks (Lucky13)
+			tls.TLS_ECDHE_RSA_WITH_RC4_128_SHA,          // Broken cipher
+
+			// all RSA-PKCS1-v1.5 ciphers are disabled - danger of Bleichenbacher attack variants
+			tls.TLS_RSA_WITH_3DES_EDE_CBC_SHA,   // Sweet32
+			tls.TLS_RSA_WITH_AES_128_CBC_SHA,    // Go stack contains (some) countermeasures against timing attacks (Lucky13)
+			tls.TLS_RSA_WITH_AES_128_CBC_SHA256, // No countermeasures against timing attacks
+			tls.TLS_RSA_WITH_AES_256_CBC_SHA,    // Go stack contains (some) countermeasures against timing attacks (Lucky13)
+			tls.TLS_RSA_WITH_RC4_128_SHA,        // Broken cipher
+
+			tls.TLS_RSA_WITH_AES_128_GCM_SHA256, // Disabled because of RSA-PKCS1-v1.5 - AES-GCM is considered secure.
+			tls.TLS_RSA_WITH_AES_256_GCM_SHA384, // Disabled because of RSA-PKCS1-v1.5 - AES-GCM is considered secure.
+		},
+		shouldFail: true,
+	},
+}
+
+func TestTLSCiphers(t *testing.T) {
+	certificate, err := getTLSCert()
+	if err != nil {
+		t.Fatalf("Unable to parse private/certificate data. %v\n", err)
+	}
+	port := getNextPort()
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprintf(w, "Hello, world")
+	})
+	server := NewServer([]string{"127.0.0.1:" + port}, handler, &certificate)
+	go func() { panic(server.Start()) }()
+	defer server.Shutdown()
+	for i, test := range tlsCipherSuitesTests {
```

Comment:
`testCases` is better name than `tlsCipherSuitesTests`, similarly `testCase` is used all over than `test`

Reply:
> testCases is better name than tlsCipherSuitesTests
IMO that's not a good idea. `testCases` is way to generic. Test vectors should look like this in Go:
https://github.com/golang/go/blob/master/src/regexp/find_test.go#L28
https://github.com/golang/go/blob/master/src/crypto/aes/aes_test.go#L134
https://github.com/golang/go/blob/master/src/html/escape_test.go#L21

> `testCase` is used all over than `test`
It's a naming thing, I change it to follow minio's style.

Reply:
> IMO that's not a good idea. testCases is way to generic. Test vectors should look like this in Go:

because `regexp` uses global variables whereas we use local variables.  Its not good to have global variable `testCases`